internal/cluster/workflow/health_check_activity.go

```diff
@@ -0,0 +1,59 @@
+// Copyright © 2020 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workflow
+
+import (
+	"context"
+
+	"k8s.io/client-go/kubernetes"
+)
+
+const HealthCheckActivityName = "k8s-health-check"
+
+type HealthCheckActivityInput struct {
+	SecretID string
+}
+
+type HealthCheckActivity struct {
+	checker       HealthChecker
+	clientFactory ClientFactory
+}
+
+// +testify:mock:testOnly=true
+
+// HealthChecker returns the result of the healthcheck
+type HealthChecker interface {
+	Check(ctx context.Context, client kubernetes.Interface) error
+}
+
+// MakeHealthCheckActivity returns a new HealthCheckActivity.
+func MakeHealthCheckActivity(
+	checker HealthChecker, clientFactory ClientFactory,
+) HealthCheckActivity {
+	return HealthCheckActivity{
+		checker:       checker,
+		clientFactory: clientFactory,
+	}
+}
+
+func (a HealthCheckActivity) Execute(ctx context.Context, input HealthCheckActivityInput) error {
+	client, err := a.clientFactory.FromSecret(ctx, input.SecretID)
+	if err != nil {
+		return err
+	}
+
+	err = a.checker.Check(ctx, client)
+	return err
```

Comment:
Recommended: unless we intend to do something with the returned error here, I think we **SHOULD** return the result of the function call immediately.

```suggestion
	return a.checker.Check(ctx, client)
```

Reply:
I agree with you.

Reply:
fixed: https://github.com/banzaicloud/pipeline/pull/3183/commits/adcd3c1b8b4fbe0b9f95f658d4118eda0953145d